controller.py

```diff
@@ -93,15 +96,19 @@ def registerAction(self, name: str, callback: Callable) -> None:
   def performAction(self, name: str, args: Optional[Tuple[any]]) -> None:
     if not name in self.actions:
       raise Exception("Actions with the name does not exist '", name, "' in: ", type(self).__name__)
-    
+
     if not callable(self.actions[name]):
       raise Exception("Registered action must be a function: ", name)
 
     self.actions[name](args=args)
-  
+
   """ Function called on controller shutdown
 
       To be overridden by classes implementing Controller if needed
   """
+  def run(self):
+      while(True):
+          t.sleep()
+
   def shutdown(self) -> None:
-    pass
\ No newline at end of file
+    pass
```

Comment:
In the shutdown function, we should set the running member variable to false, and join the thread with
`self.thread.join()`